src/features/selectedPost/selectedPostSlice.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+// eslint-disable-next-line import/no-cycle
+import { Post } from '../../types/Post';
+
+const initialState = {
+  selectedPost: null as null | Post,
+};
+
+export const selectedPostSLice = createSlice({
```

Comment:
The slice name `selectedPostSLice` should be renamed to `selectedPostSlice` to follow JavaScript naming conventions.